src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -5476,7 +5476,58 @@ LogicalResult ONNXUniqueOp::inferShapes(
 
 LogicalResult ONNXUpsampleOp::inferShapes(
     std::function<void(mlir::Region &)> doShapeInference) {
-  return emitError(NOT_IMPLEMENTED_MESSAGE);
+  // Sanity checks on input data argument
+  if (!X().getType().isa<RankedTensorType>()) {
```

Comment:
Could the sanity check be put into verifier?

Reply:
Good point. I've added a verifier, and put the checks in there